File: frontend/console/src/api/logic-device.js
import Axios from "@/http";

export function createLogicDevice(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-device', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveLogicDevice(logicDeviceId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/logic-device/${ logicDeviceId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateLogicDevice(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/logic-device', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicDevice(logicDeviceId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/logic-device/${ logicDeviceId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicDeviceWithFlag(logicDeviceId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/logic-device/${ logicDeviceId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllLogicDevices() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/logic-device/all').then(res => {
            resolve(res)
        })
    })
}

export function queryLogicDevice (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-device/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/logic_device.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_LOGIC_DEVICE = "/config/logic-device"
const URL_ROUTER_LOGIC_DEVICE_BY_ID = "/config/logic-device/:logic-device-id"
const URL_ROUTER_LOGIC_DEVICE_QUERY = "/config/logic-device/query"
const URL_ROUTER_LOGIC_DEVICE_ALL = "/config/logic-device/all"

func LogicDeviceRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_LOGIC_DEVICE, config.CreateLogicDevice)
	r.GET(URL_ROUTER_LOGIC_DEVICE_BY_ID, config.RetrieveLogicDevice)
	r.PUT(URL_ROUTER_LOGIC_DEVICE, config.UpdateLogicDevice)
	r.DELETE(URL_ROUTER_LOGIC_DEVICE_BY_ID, config.DeleteLogicDevice)
	r.POST(URL_ROUTER_LOGIC_DEVICE_QUERY, config.QueryLogicDevice)
	r.GET(URL_ROUTER_LOGIC_DEVICE_ALL, config.GetAllLogicDevices)
	return r
}